server/controllers/loginController.js

```
@@ -1,47 +1,58 @@
+//handel errors
+
 const { compare } = require('bcrypt');
-const {sign} = require('jsonwebtoken');
+const { sign } = require('jsonwebtoken');
 const { checkEmailQuery } = require('../database/queires');
 const { signinSchema } = require('../validation');
+const customError = require('../utils/error/custamizeError')
+
+
 
 const privateKey = process.env.SECRET_KEY;
 
-const login = (req, res)=>{
+const login = (req, res, next) => {
+    let id = '';
+    const { password, email } = req.body;
     signinSchema.validateAsync(req.body)
-    .then(value =>{
-        let id = ''
-        const {email, password} = value;
-        checkEmailQuery(email)
-            .then(data => {
-                if(data.rowCount === 0){
-                    res.status(400).json({msg: 'The email does not exist! signup insted'})
-                }else{
-                    const {password:hashedPassword, id, email} = data.rows[0];
-                    id = data.rows[0].id
-                    return compare(password, hashedPassword)
-                    
-                }
-            }).then(isMatched =>{
-            if(isMatched){
-                sign({id, email}, privateKey, (err, token)=>{
-                    if(err){
-                        res.status(500).json({msg: 'Internal server error'})
-                    }else{
+        .then(() => checkEmailQuery(email))
+        .then(data => {
+            if (data.rowCount === 0) {
+                throw customError('The email does not exist! signup insted', 401)      //throw err
+            } else {
+                id = data.rows[0].id
+                const { password: hashedPassword } = data.rows[0];
+                return compare(password, hashedPassword)
+            }
+        })
+        .then(isMatched => {
+            if (isMatched) {
+                sign({ id, email }, privateKey, (err, token) => {
+                    if (err) {
+                        res.status(500).json({ message: 'Internal server error' })
```

You can throw an error here like the others. The same for the other controllers handle their errors as you did here

---

yes sure i fixed it